File: programmers/etc/level2/전화번호_목록.py
def solution(phone_book):
    phone_book = sorted(phone_book, key=len)
    s = set()
    for p in phone_book:
        for i in range(1, len(p)):
            if p[:i] in s:
                return False
        s.add(p)
    return True


phone_book = ["119", "97674223", "1195524421"]
print(solution(phone_book)) # False

phone_book = ["123","456","789"]
print(solution(phone_book)) # True

phone_book = ["12","123","1235","567","88"]
print(solution(phone_book)) # False

File: Java/programmers/etc/level2/전화번호목록.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 전화번호목록 {
    public boolean solution(String[] phone_book) {
        Arrays.sort(phone_book);
        for (int i=0; i<phone_book.length-1; i++) {
            if (phone_book[i+1].startsWith(phone_book[i])) {
                return false;
            }
        }
        return true;
    }
}
